server.js

```diff
@@ -33,56 +33,71 @@ app.post('/book', createBook);
-  const filterQuery = {};
-  // change location to email?
-  if (request.query.location) {
-    filterQuery.location = request.query.location;
+
+ async function getBooks(request, response)  {
+  try {
+    const books = await Book.find({email: request.query.email});
+    response.send(books);
+  }
+  catch (error) {
+    console.error(error);
+    response.status(400).send('Could not find books');
   }
-  // const books = await BookModel.find(filterQuery);
-  const books = await Book.find(filterQuery);
+};
 
-  response.send(books);
-});
 
-app.post('/books', async (request, response) => {
-  const newBook = await Book.create({
-    ...request.body,
-    email: request.query.email,
-  });
-  response.send(newBook);
-});
 
-app.delete('/books/:id', async (request, response) => {
-  // if email sent on query, then...
-  if (request.query.email) {
-    // get the book that matches id AND email
-    const foundBook = await Book.findOne({
-      _id: request.params.id,
-      email: request.query.email,
-    });
-
-    // only delete if found
-    console.log({ foundBook });
-
-    // const deleteResult = await Book.findByIdAndDelete(request.params.id);
-
-    if (foundBook) {
-      const deleteResult = await Book.findByIdAndDelete(request.params.id);
-      console.log({ deleteResult });
-      response.status(204).send('success');
-    } else {
-      response.status(400).send('You cannot delete that book'); // correct status code??
+async function createBook(request, response)  {
+  try {
+    const book = await Book.create(request.body);
+    response.send(book)
+  }
+
+  catch (error) {
+  console.error(error);
+  response.status(400).send('Unable to create book');
+  }
+};
+
+async function deleteBook(request, response) {
+  try {
+    const email = request.query.email;
+    const id = request.params.id;
+    const book = await Book.findOne({_id: id,email});
+    if (!book){
+      response.status(400).send('Unable to delete book');
+      return;
+    }
+    if (book.email !== email) {
+      response.status(400).send('Unable to delete book');
+      return;
     }
-  } else {
-    response.status(403).send('You must be logged in');
+    await Book.findByIdAndDelete(id);
+    response.send('Success!');
   }
-});
 
-app.put('/books/:id', async (request, response) => {
+  catch (error) {
+    console.error(error);
+    response.status(400).send('Unable to delete book');
+  }
+}
+
+async function updateBook (request, response)  {
   const id = request.params.id;
-  let updatedBook = await Book.findByIdAndUpdate(id, { ...request.body });
-  response.status(204).send(updatedBook);
-});
+  const email = request.query.email;
+  try {
+    const bookToUpdate = await Book.findOne({_id: id, email});
+    if (!bookToUpdate) {
+      response.status(400).send('Unable to update book');
+      return
+    }
+    const updatedBook = await Book.findByIdAndUpdate(id, request.body, {new: true});
+    response.send(updateBook);
```

Comment:
Nice work using async await and handling errors!

- For line 95 , you will want to res.status(200).send(updateBook) 👍